ccl/multitenantccl/tenantcostclient: TestSQLLivenessExemption flakes

**Describe the problem**

The test seems to fail as a side-effect of https://github.com/cockroachdb/cockroach/pull/87533. 

See https://github.com/cockroachdb/cockroach/blob/59a70ba8eb11cd3ce708ca85affc112a26b8741a/pkg/ccl/multitenantccl/tenantcostclient/tenant_side_test.go#L932-L933

In retrospect, I think setting the timeout to be on the order of the heartbeat interval was a mistake. It should probably be something related to the timeout duration. Maybe some combination thereof.

Was able to replicate locally when run with `--stress`. 
```
57 runs completed, 1 failures, over 7s
context canceled
```
```
--- FAIL: TestSQLLivenessExemption (0.63s)
    test_log_scope.go:161: test logs captured to: /Users/aadityasondhi/go/src/github.com/cockroachdb/cockroach/tmp/_tmp/cd6b537d257847226387a72b864251e8/logTestSQLLivenessExemption1538851179
    test_log_scope.go:79: use -show-logs to present logs inline
    test_server_shim.go:417: constructing session: operation "sqlliveness create session" timed out after 61ms (given timeout 50ms): could not insert session 53564e4363224d14a47171adef61ad85: aborted during DistSender.Send: context deadline exceeded
```

--- a/pkg/ccl/multitenantccl/tenantcostclient/tenant_side_test.go
+++ b/pkg/ccl/multitenantccl/tenantcostclient/tenant_side_test.go
@@ -929,8 +929,7 @@ func TestSQLLivenessExemption(t *testing.T) {
 	st := cluster.MakeTestingClusterSettings()
 	// Make the tenant heartbeat like crazy.
 	ctx := context.Background()
-	//slinstance.DefaultTTL.Override(ctx, &st.SV, 20*time.Millisecond)
-	slinstance.DefaultHeartBeat.Override(ctx, &st.SV, 50*time.Millisecond)
+	slinstance.DefaultHeartBeat.Override(ctx, &st.SV, 10*time.Millisecond)
 
 	_, tenantDB := serverutils.StartTenant(t, hostServer, base.TestTenantArgs{
 		TenantID:                    tenantID,

--- a/pkg/sql/sqlliveness/slinstance/slinstance.go
+++ b/pkg/sql/sqlliveness/slinstance/slinstance.go
@@ -260,9 +260,12 @@ func (l *Instance) heartbeatLoop(ctx context.Context) {
 		case <-t.C:
 			t.Read = true
 			s, _ := l.getSessionOrBlockCh()
+			// TODO(aaditya): consider combining `DefaultTTL` and `DefaultHeartBeat` into a single knob to make these
+			//  timeouts less fragile
+			timeout := l.ttl()/2 + l.hb()
 			if s == nil {
 				var newSession *session
-				if err := contextutil.RunWithTimeout(ctx, "sqlliveness create session", l.hb(), func(ctx context.Context) error {
+				if err := contextutil.RunWithTimeout(ctx, "sqlliveness create session", timeout, func(ctx context.Context) error {
 					var err error
 					newSession, err = l.createSession(ctx)
 					return err
@@ -284,7 +287,7 @@ func (l *Instance) heartbeatLoop(ctx context.Context) {
 				continue
 			}
 			var found bool
-			err := contextutil.RunWithTimeout(ctx, "sqlliveness extend session", l.hb(), func(ctx context.Context) error {
+			err := contextutil.RunWithTimeout(ctx, "sqlliveness extend session", timeout, func(ctx context.Context) error {
 				var err error
 				found, err = l.extendSession(ctx, s)
 				return err

--- a/pkg/sql/sqlliveness/slinstance/slinstance_test.go
+++ b/pkg/sql/sqlliveness/slinstance/slinstance_test.go
@@ -192,7 +192,7 @@ func TestSQLInstanceDeadlinesExtend(t *testing.T) {
 	}
 	defer cleanUpFunc()
 	// advance manual clock so that session expires
-	mt.Advance(20 * time.Millisecond)
+	mt.Advance(slinstance.DefaultTTL.Get(&settings.SV))
 
 	// expect session to expire
 	require.Eventually(

